src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -0,0 +1,252 @@
+    }
+
+    /**
+     * Get the Context menu anchor position
+     * We calculate the achor coordinates from measureInWindow async method
+     *
+     * @returns {Promise<Object>}
+     */
+    getContextMenuMeasuredLocation() {
+        return new Promise((resolve) => {
+            if (this.contextMenuAnchor) {
+                this.contextMenuAnchor.measureInWindow((x, y) => resolve({x, y}));
+            } else {
+                resolve({x: 0, y: 0});
+            }
+        });
+    }
+
+    isActiveReportAction(actionID) {
+        return this.state.reportAction.reportActionID === actionID;
+    }
+
+    /**
+     * Show the ReportActionContextMenu modal popover.
+     *
+     * @param {Object} [event] - A press event.
+     * @param {string} [selection] - A copy text.
+     * @param {Element} contextMenuAnchor - popoverAnchor
+     * @param {Number} reportID - Active Report Id
+     * @param {Object} reportAction - ReportAction for ContextMenu
+     * @param {String} draftMessage - ReportAction Draftmessage
+     * @param {Function} [onShow=() => {}] - Run a callback when Menu is shown
+     * @param {Function} [onHide=() => {}] - Run a callback when Menu is hidden
+     */
+    showContextMenu(
+        event,
+        selection,
+        contextMenuAnchor,
+        reportID,
+        reportAction,
+        draftMessage,
+        onShow = () => {},
+        onHide = () => {},
+    ) {
+        const nativeEvent = event.nativeEvent || {};
+        this.contextMenuAnchor = contextMenuAnchor;
+        this.onPopoverHide = onHide;
+        this.getContextMenuMeasuredLocation().then(({x, y}) => {
+            this.setState({
+                cursorRelativePosition: {
+                    horizontal: nativeEvent.pageX - x,
+                    vertical: nativeEvent.pageY - y,
+                },
+                popoverAnchorPosition: {
+                    horizontal: nativeEvent.pageX,
+                    vertical: nativeEvent.pageY,
+                },
+                reportID,
+                reportAction,
+                selection,
+                isPopoverVisible: true,
+                reportActionDraftMessage: draftMessage,
+            }, onShow);
+        });
+    }
+
+    /**
+     * This gets called on Dimensions change to find the anchor coordinates for the action context menu.
+     */
+    measureContextMenuAnchorPosition() {
+        if (!this.state.isPopoverVisible) {
+            return;
+        }
+        this.getContextMenuMeasuredLocation().then(({x, y}) => {
+            if (!x || !y) {
+                return;
+            }
+            this.setState(prev => ({
+                popoverAnchorPosition: {
+                    horizontal: prev.cursorRelativePosition.horizontal + x,
+                    vertical: prev.cursorRelativePosition.vertical + y,
+                },
+            }));
+        });
+    }
+
+    runAfterContextMenuHide() {
```

Comment:
NAB because both names are bad:
```suggestion
    runAndResetOnPopoverHide() {
```